### DISCUSSION TITLE
Make the option to extend the day more flexible

### DISCUSSION THREAD

**Discussion by cadadr** [iSoron/uhabits#1617] [Feature Requests]
As someone with irregular sleep, the option to extend day beyond midnight is one that i really like. I use checkmark widgets a lot for the basics of my daily routine and they staying checked until my day is really complete is both phsychologically and organisationally helpful.

But my sleep pattern changes a lot and often it's more than three hours after midnight that my day ends.

So it would be really helpful if it was possible to change how many hours after midnight the day was extended.

I see that this might complicate implementation somewhat if it was completely unbounded, but I suppose a choice between 0-6 or 0-9 hours would accommodate most people, if feasible.

**Comment by Farzat07** [comment]
Would be nice if the number could be negative, for people whose days start before midnight.

For example, in my culture the day technically ends at sunset. I sometimes want to add an entry but want it to count towards the next day.

**Comment by ghost** [comment]
I second this, would love this.
